src/dos/drives.cpp

```diff
@@ -25,20 +25,32 @@
 
 extern char sfn[DOS_NAMELENGTH_ASCII];
 
+constexpr bool is_special_character(const char c)
+{
+	constexpr auto special_characters = std::string_view("\"+=,;:<>[]|?*");
+	return special_characters.find(c) != std::string_view::npos;
+}
+
 // TODO Right now label formatting seems to be a bit of mess, with various
 // places in code setting/expecting different format, so simple GetLabel() on
 // a drive object might not yield an expected result. Not sure how to sort it
 // out, but it will require some attention to detail.
-// Also: this function is too strict - it removes all punctuation when *some*
-// punctuation is acceptable in drive labels (e.g. '_' or '-').
 //
 std::string To_Label(const char* name) {
 	// Reformat the name per the DOS label specification:
 	// - Upper-case, up to 11 ASCII characters
 	// - Internal spaces allowed but no: tabs ? / \ | . , ; : + = [ ] < > " '
 	std::string label(name);
 	trim(label); // strip front-and-back white-space
-	strip_punctuation(label); // strip all punctuation
+	label.erase(std::remove_if(label.begin(),
+	                           label.end(),
+	                           [](unsigned char c) {
+		                           return c == '\t' || c == '/' ||
+		                                  c == '\\' || c == '.' ||
+		                                  c == '\'' ||
+		                                  is_special_character(c);
+	                           }),
+	            label.end());
```

Comment:
According to this, period characters should be allowed, no?

https://ftp.zx.net.nz/pub/archive/ftp.microsoft.com/MISC/KB/en-us/26/295.HTM

Also, this comment seems to be slightly wrong:

```
Internal spaces allowed but no: tabs ? / \ | . , ; : + = [ ] < > " '
```

The single-quote char is allowed, so it should be removed from the end.

I'd suggest putting this link here for reference:
https://ftp.zx.net.nz/pub/archive/ftp.microsoft.com/MISC/KB/en-us/26/295.HTM

Reply:
> According to this, period characters should be allowed, no?

Then the comment is wrong about the period, too.

Reply:
Something important that this article mentions is that control characters are not allowed. I'll need to add that.

Reply:
Cool, let's make sure everything is in sync (both the code and comments) according to that Microsoft knowledge base article. For extra points you cross-check with real MS-DOS 6.22 running in a VM.